tools/wptrunner/wptrunner/executors/base.py

```diff
@@ -76,11 +77,45 @@ def __call__(self, test, result, extra=None):
 testharness_result_converter = TestharnessResultConverter()
 
 
+def hash_screenshot(data):
+    """Computes the sha1 checksum of a base64-encoded screenshot."""
+    return hashlib.sha1(base64.b64decode(data)).hexdigest()
+
+
+def _convert_reftest_screenshots(extra):
+    """Converts reftest_screenshots field into the following format:
+    {"path": {"hash": "sha1:...", "screenshot": "base64-encoded data"}, ...}
+    """
+    log_data = extra.get("reftest_screenshots")
+    if not log_data:
+        return None
+    screenshots = {}
+    for item in log_data:
+        if type(item) != dict:
+            # Skip relation strings.
+            continue
+        if "hash" in item:
+            hash_val = item["hash"]
+        else:
+            # Marionette internal reftest runner does not produce hash.
+            hash_val = hash_screenshot(item["screenshot"])
+        screenshots[strip_server(item["url"])] = {
+            # Prepend the hash method to checksum to satisfy wpt.fyi.
+            "hash": "sha1:" + hash_val,
+            "screenshot": item["screenshot"],
+        }
+    return screenshots
+
+
 def reftest_result_converter(self, test, result):
+    extra = result.get("extra", {})
+    screenshots = _convert_reftest_screenshots(extra)
+    if screenshots:
+        extra["screenshots"] = screenshots
```

Comment:
Note that I think this will modify `result["extra"]` in place. I don't think it'd cause problems in this case, but it could be dangerous.